crates/pallet-domains/src/lib.rs

```diff
@@ -556,11 +556,31 @@ impl<T: Config> Pallet<T> {
     ) -> Result<(), BundleError> {
         let current_block_number = frame_system::Pallet::<T>::current_block_number();
 
-        if let Some(last_finalized) =
-            current_block_number.checked_sub(&T::ConfirmationDepthK::get())
-        {
-            if bundle.header.primary_number <= last_finalized {
-                return Err(BundleError::StaleBundle);
+        // Reject the stale bundles so that they can't be used by attacker to occupy the block space without cost.
+        let confirmation_depth_k = T::ConfirmationDepthK::get();
+        if let Some(finalized) = current_block_number.checked_sub(&confirmation_depth_k) {
+            {
+                // Ideally, `bundle.header.primary_number` is `current_block_number - 1`, we need
+                // to handle the edge case that `T::ConfirmationDepthK` happens to be 1.
+                let is_stale_bundle = if confirmation_depth_k.is_zero() {
+                    unreachable!(
+                        "ConfirmationDepthK is guaranteed to be non-zero at genesis config"
+                    )
+                } else if confirmation_depth_k == One::one() {
+                    bundle.header.primary_number < finalized
+                } else {
+                    bundle.header.primary_number <= finalized
```

Comment:
I could be missing something. But can't we enforce the same cutoff when K depth is >= 1?

Reply:
Sorry. Used > instead of <. Anyway

My question was. why not make both Kdepth == 1 and Kdepth > 1 has same check
`bundle.header.primary_number <= finalized`

Reply:
I see. The background of the stale bundle check https://github.com/subspace/subspace/issues/1137#issue-1566726703 was basically to prevent an attacker from resubmitting any old valid bundles again to occupy the block space without any cost, since #1056 is expected to take care of the last K blocks in the range of [N-K + 1, N] (both inclusive), then #1147 needs to take care of the blocks older than N-K (inclusive), thus it ought to be `<=` in theory.